### ISSUE TITLE
CDC  like  support 

### ISSUE THREAD

**Issue by rongfengliang** [castledio/castled#45]
I had do some tests with docker-compose .  will support CDC like  features?  current we  must run pipeline with  regularly task

but MessageSyncStats is not persistent. maybe not  god for large message transform

**Comment by franklinio** [comment]
Castled supports full and incremental syncs. Incremental is essentially same as CDC. After the first sync, all the future pipeline runs will capture the delta (or diff) in the source table compared to its previous run and only transfer the change.  Of course, this is all done without persisting any table data at our end. 

For e.g if that table is a static one, only the first pipeline will sync the data, all the pipeline runs after that will be a no-op as there is nothing to update.

**Comment by rongfengliang** [comment]
@franklinio  I want to write one datawarehouse driver for castled (with dremio). currently I'm still looking for source code how diff works.   any documents that can help ?

**Comment by aruntcet2010** [comment]
@rongfengliang We are working on the code level documentation. Meanwhile you can check the implementations of [WarehouseDataPoller](https://github.com/castledio/castled/blob/main/connectors/src/main/java/io/castled/warehouses/WarehouseDataPoller.java) interface to check how the delta computation is done(eg: [SnowflakeDataPoller](github.com/castledio/castled/blob/main/connectors/src/main/java/io/castled/warehouses/connectors/snowflake/SnowflakeDataPoller.java)).

Basically, what we do is we create a snapshot table in a custom schema used for Castled to snapshot the records which have been synced so far and use that as the reference to compare it with the query results when the pipeline is run the next time.

**Comment by rongfengliang** [comment]
@franklinio  Ok, thanks.
